src/checkers/game.rs

```diff
@@ -0,0 +1,457 @@
+use checkers::ai;
+
+use checkers::Board;
+use checkers::BoardPosition;
+use checkers::Direction;
+use checkers::JumpMove;
+use checkers::KingPiece;
+use checkers::ManPiece;
+use checkers::OccupiedTile;
+use checkers::PieceType;
+use checkers::Player;
+use checkers::SimpleMove;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum GameState {
+	/// The game has not yet finished
+	InProgress,
+	
+	/// The game has finished. The `u32` is the ID of the winning player.
+	GameOver{ winner_id : u32 }
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum MoveError {
+	InvalidMove,
+	ShouldHaveJumped
+}
+
+struct PlayerInfo {
+	player : Player,
+	direction : Direction
+}
+
+pub struct Game {
+	players : [PlayerInfo; 2],
+	board : Board,
+	
+	current_player_index : usize,
+	
+	available_simple_moves : Vec<SimpleMove>,
+	available_jump_moves : Vec<JumpMove>
+}
+
+impl Game {
+	pub fn new() -> Game {
+		let (player1, player2) = Game::two_players();
+		
+		let board = Board::new_checkerboard(&player1, &player2);
+		
+		Game::with_board_and_players(board, player1, player2)
+	}
+	
+	/// Creates a new Checkers game with an 8x8 board and the specified piece
+	/// layout `player1_positions` contains the pieces for the first player,
+	/// and `player2_positions` contains the pieces for the second player. All
+	/// pieces are initially `ManPiece`s.
+	///
+	/// # Panics
+	///
+	/// Panics if any element in *player1_positions* or *player2_positions*
+	/// is a `BoardPosition` with a row or column outside the range [0, 7].
+	/// Also panics if any two positions are exactly the same.
+	#[cfg(test)]
```

Comment:
Yes. You should try using this function from the main function, and see what happens. It won't even compile, if I remember right.

I don't think it would be harmful to expose this function to non-test builds, but we don't really need it except for tests, so why bother?